lib/_http_outgoing.js

```diff
@@ -426,6 +430,37 @@ OutgoingMessage.prototype.getHeader = function getHeader(name) {
 };
 
 
+// Returns an array of the names of the current outgoing headers.
+OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
+  return (this._headers ? Object.keys(this._headers) : []);
+};
+
+
+// Returns a shallow copy of the current outgoing headers.
```

Comment:
I agree it is different. I just want it stated in the docs, rather than "implied" (your word). I suggest you include what you just wrote:

> The array values can be mutated, which is useful because you do not have to continually execute setHeader(), which has additional overhead, when you want to add more values for that header.

You seem dead set against adding an explanatory sentence, because you feel the user should be able to derive this understanding from the word "shallow" (if I understand you correctly), but I don't think one extra sentence is overkill here.